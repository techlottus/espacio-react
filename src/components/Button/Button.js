

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


