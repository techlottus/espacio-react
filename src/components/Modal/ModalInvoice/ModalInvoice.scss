@import '../../../App.scss';
@import '../Modal.scss';

.modal-invoice {
  @extend .modal-container;
  font-family: 'Nunito', sans-serif;

  .modaltextinvoice {
    margin-bottom: pixeltorem(16px);  
    font-family: 'Nunito', sans-serif;
  }

  .modaltextstickinvoice {
    font-size: pixeltorem(12px);
    line-height: pixeltorem(16px);
    font-family: 'Nunito', sans-serif;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      font-size: pixeltorem(14px);
      
    }
  }

}