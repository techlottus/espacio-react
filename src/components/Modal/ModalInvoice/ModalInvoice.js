import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalBackInvoice,
  btnModalNextInvoice,
  optionsStates,
} from "../../../constants/Payment.constant";
import "./ModalInvoice.scss";
import { formatString } from "../../../helpers/formatString";
import { useSelector } from "react-redux";

export const ModalInvoice = ({ isShow, onClose, saveModal, values, outside }) => {
  const { texts } = useSelector(state => state.texts);
  const [isModal, setIsModal] = useState(false);
  const [stateText, setStateText] = useState(null)

  const modalRef = useRef();

  useEffect(() => {
    setIsModal(isShow);
    if (isShow) {
      setStateText(optionsStates.filter((state) => values.state === state.value)[0].text) 
    }
  }, [isShow, values]);

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-invoice">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
          <span className="material-icons icon iconright">{texts?.payment.modals.modalIcon}</span>
            <div className="title">{texts?.payment.modals.modalInvoice.title}</div>
          </div>
          <div className="modalbody">
            <p> {values.rfc ? `${texts?.payment.cardInvoice.rfc} ${values.rfc.toUpperCase()}` : ""}</p>
            <p>
              {values.businessName
                ? `${texts?.payment.cardInvoice.businessName} ${values.businessName}`
                : ""}
            </p>
            <p>
              {values.fullName ? `${texts?.payment.cardInvoice.name} ${values.fullName}` : ""}
            </p>
            <p>
              {values.zipCode
                ? `${texts?.payment.cardInvoice.address} ${values.street}, ${values.neighborhood}, ${values.city}, ${formatString(stateText)}, ${values.zipCode}`
                : ""}
            </p>
            <p> {values.email ? `${texts?.payment.cardInvoice.email} ${values.email}` : ""}</p>
            <p> {values.cfdiUsage ? `${texts?.payment.cardInvoice.cfdi} ${values.cfdiUsage}` : ""}</p>
            <p> {values.regime ? `${texts?.payment.cardInvoice.regime} ${values.regime}` : ""}</p>
            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalBackInvoice, 
                    title: texts?.payment.btnModal.back,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onClose();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalNextInvoice,
                    title: texts?.payment.btnModal.next,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    saveModal();
                    onClose();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
