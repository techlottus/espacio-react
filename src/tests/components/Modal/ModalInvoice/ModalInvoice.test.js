import React, { createRef } from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ModalInvoice } from "../../../../components/Modal/ModalInvoice/ModalInvoice";
import { Button } from "../../../../components/Button/Button";
import { act } from "react-dom/test-utils";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
    texts: {},
}

const isShow = true;
const onClose = jest.fn().mockImplementation(() => {});
const saveModal = true;
const values = {state:"02"}
const outside = jest.fn();


let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <ModalInvoice isShow={isShow} onClose={onClose} saveModal={saveModal} values={values} outside={outside} />
    </Provider>
  );
};

describe("Test <ModalInvoice/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("should close modal", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).first().props().onClick();
    });
    wrapper.update();
  });

});
