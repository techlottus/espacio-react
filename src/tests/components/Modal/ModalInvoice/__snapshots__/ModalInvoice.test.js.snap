// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ModalInvoice/> should init component 1`] = `
<div
  className="modal-invoice"
>
  <div
    className="modalcontent"
  >
    <div
      className="modalheader"
    >
      <span
        className="material-icons icon iconright"
      >
        help
      </span>
      <div
        className="title"
      />
    </div>
    <div
      className="modalbody"
    >
      <p>
         
      </p>
      <p />
      <p />
      <p />
      <p>
         
      </p>
      <p>
         
      </p>
      <div
        className="modalbtns"
      >
        <div
          className="btnsecundary"
        >
          <lottus-button />
        </div>
        <div
          className="btnprimary"
        >
          <lottus-button />
        </div>
      </div>
    </div>
  </div>
</div>
`;
